### DISCUSSION TITLE
Capture the output of cue vet command

### DISCUSSION THREAD

**Discussion by Roohiya** [cue-lang/cue#1866] [Q&A]
Trying to capture the result of cue vet command that validates a file. Currently I am trying to run the command in Python's subprocess.run function that runs the following bash commands from a file:

```
#!/usr/bin/env bash

cd $1
cue vet $2 $3
```

But I am trying to get the output if the validation fails with process.stdout etc, but I am unable to capture the terminal logs. Any ideas?

**Comment by verdverm** [comment]
Subprocess has many options: https://docs.python.org/3/library/subprocess.html

This SO may be helpful: https://stackoverflow.com/questions/41171791/how-to-suppress-or-capture-the-output-of-subprocess-run

**Comment by Roohiya** [comment]
Hi yes I have been trying subprocess with no avail. 

I have tried these few methods:

Case 1
```process = subprocess.run([<above bash script>, arg1, arg2, arg3], stdout=subprocess.PIPE)
     print(process.stdout)
``` 

Output
```b''```


Case 2
```process = subprocess.Popen([<above bash script>, arg1, arg2, arg3], stdout=subprocess.PIPE)
     output, error = process.communicate()
     print(output)
```
Output
```b''```

Basically the cue vet command can't seem to be captured even with the stdout

**Comment by slewiskelly** [comment]
`cue vet` outputs to `stderr` not `stdout`. So you'll have to capture that.

```console
# Redirect stdout to output file, still displays to terminal.
$ cue vet -t data='{"foo": "baz"}' -d '#file' foo.cue foo.json > out.txt
foo: conflicting values "baz" and "bar":
    ./foo.json:2:12
    json:1:9

# Output file is empty.
$ wc out.txt
       0       0       0 out.txt

# Redirect stderr to output file, nothing displayed to terminal.
$ cue vet -t data='{"foo": "baz"}' -d '#file' foo.cue foo.json 2> out.txt

# Outpul file has contents.
$ wc out.txt
       3       8      74 out.txt

$ Output file containts what is expected.
$ cat out.txt
foo: conflicting values "baz" and "bar":
    ./foo.json:2:12
    json:1:9
````

**Comment by mvdan** [comment]
This seems like a CUE-related question rather than a bug or feature request, so I'll move it to a discussion.
